go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -116,47 +122,81 @@ type database struct {
 }
 
 func (d *database) SetOptions(cnOptions map[string]string) error {
-	if d.uri.Scheme == "grpc+tls" {
-		d.creds = credentials.NewTLS(&tls.Config{})
-	} else {
-		d.creds = insecure.NewCredentials()
-	}
+	var tlsConfig tls.Config
 
-	if val, ok := cnOptions[OptionSSLSkipVerify]; ok && val == adbc.OptionValueEnabled {
-		if d.uri.Scheme != "grpc+tls" {
+	mtlsCert := cnOptions[OptionMTLSCertChain]
+	mtlsKey := cnOptions[OptionMTLSPrivateKey]
```

Comment:
is an empty string a valid value for either of this? You could use the form
`mtlsCert, tlsCertOK := cnOptions[OptionMTLSCertChain]` to see if the key existed but an empty string was provided (the second return is a boolean saying whether the key existed in the map). Same for `mtlsKey`

Reply:
I don't believe so, mTLS requires a valid cert and corresponding private key